javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -232,6 +232,23 @@ public Map<Request, List<ProxyMethod>> map(Operation operation) {
             ProxyMethod proxyMethod = builder.build();
             proxyMethods.add(proxyMethod);
 
+            if(asyncRestResponseReturnType instanceof GenericType
+                    && ((GenericType) asyncRestResponseReturnType).getTypeArguments()[0] instanceof GenericType
+                    && ((GenericType) ((GenericType) asyncRestResponseReturnType).getTypeArguments()[0]).getName().equals("ResponseBase")) {
+                IType asyncResponseWithNoHeaders = getAsyncRestResponseReturnType(operation, responseBodyType,
+                        settings.isDataPlaneClient(), settings, true);
+                builder.returnType(asyncResponseWithNoHeaders);
+                builder.name(operationName + "NoCustomHeaders");
```

Comment:
Do we want a flag for it? Do we expect it be direct exposed to user through sync/async wrapper?